### DISCUSSION TITLE
How to build a Docker image using github action for the last commit on a pull request

### DISCUSSION THREAD

**Discussion by uditgaurav** [community/community#25382] [Actions]
<p>Hi all,</p>
<ul>
<li>I was trying to build a Docker Image from a Dockerfile file in my repository using/inside the GitHub action.</li>
<li>The image should be created for the last commit on a pull request.</li>
<li>I was running the Github Actions on <code>issue_comment</code> basis which also involves pull request comments and I tried to use <code>docker build</code> command inside this like:</li>
</ul>
<pre><code class="lang-auto">name: Litmus-CI
on:
  issue_comment:
    types: [created, edited]
jobs:
  tests:
    if: contains(github.event.comment.html_url, '/pull/') &amp;&amp; contains(github.event.comment.body, '/run-e2e')
    runs-on: ubuntu-latest
    steps:

      - name: Running git checkout
        uses: actions/checkout@v2

      - name: Build docker image
        run: |
          sudo docker build . -f build/ansible-runner/Dockerfile -t uditgaurav/demo-litmus-e2e:v1
          sudo docker images

</code></pre>
<p>So will the image build using the above code will be on the last commit on pull request basis or it will be a general image build of the repository?<br>
secondly, Are there any actions around this - for building the image of the last commit in PR?<br>
Thanks!</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>The <code>GITHUB_SHA</code> of <code>issue_comment</code> event is the Last commit on default branch. So it could not get the last commit on pull_request.<br>
As a work around , please follow next steps to get the last commit in PR</p>
<ol>
<li>You could get the PR number from the payload of issue comment event : <code>github.event.issue.number</code> .</li>
<li>Use the PR number to get its commits through <a href="https://developer.github.com/v3/pulls/#list-commits-on-a-pull-request" rel="nofollow noopener">this</a> API  . You could use <a href="https://github.com/octokit/request-action" rel="nofollow noopener">octokit/request-action</a> to send request to GitHub API.  This action will output the response body to an output variable data.</li>
<li>Use <code>jq</code> to get the commit id from the response data. Then set the value to an output variable</li>
<li>When execute checkout action, you could add <code>ref</code> parameter with the commit id as value.<br>
There is my sample workflow :</li>
</ol>
<pre><code class="lang-auto">on:
  issue_comment:
    types: [created, edited]
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: octokit/request-action@v2.x
        id: get_PR_commits
        with:
          route: GET /repos/:repo/pulls/:pull_number/commits
          repo: ${{ github.repository }}
          pull_number: ${{ github.event.issue.number }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      - name: set commit to output
        id: getcommit
        run: | 
           prsha=$(echo $response | jq '.[-1].sha'  | tr -d '"')
           echo "::set-output name=sha::$prsha" 
        env: 
          response:  ${{ steps.get_PR_commits.outputs.data }}
          
      - uses: actions/checkout@v2
        with:
          ref: ${{steps.getcommit.outputs.sha}}  
</code></pre>

**Comment by uditgaurav** [comment]
<p>This works fine<br>
Awesome Thanks!</p>
